### PR TITLE
icmp: support IPv6 addresses

### PR DESCRIPTION
**Pull request by stapelberg**
For hosts which have both an A and an AAAA record, the default stays
IPv4, so there’s no change in behavior.

In addition to being able to specify IPv4 addresses, you can now specify
IPv6 addresses. Tested using curl
http://localhost:9115/probe?module=ping_1s&target=127.0.0.1 and
http://localhost:9115/probe?module=ping_1s&target=::1

related to #24

### CODE REVIEW

File: icmp.go [hunk 1]
```diff
@@ -40,24 +42,38 @@ func getICMPSequence() uint16 {
...
 		return
 	}
-	defer socket.Close()
 
-	ip, err := net.ResolveIPAddr("ip4", target)
+	var (
+		socket                 *icmp.PacketConn
+		requestType, replyType icmp.Type
+	)
+	if ip.IP.To4() == nil {
```
Comment (brian-brazil) [inline icmp.go]: We should export whether v4 or v6 is being used

Comment (stapelberg) [inline icmp.go]: Do you have a suggestion as to how to export that?

Comment (brian-brazil) [inline icmp.go]: As a gauge with either the value 4 or 6.

**Comment by hasso** [comment]
While I like the code, I disagree with logic – there is no way to probe by name and force it to IPv6. I borrowed some of your code to enhance the patch I'm using here quite some time already and which adds support for forcing probes to IPv4 or IPv6 to all modules - #64 .

**Comment by stapelberg** [comment]
Alright then. Closing this one in favor of your PR #64

**Pull request closed without merging**